File: python/src/code_smells/dispensables/dead_code.py
"""
Dead Code Code Smell Example

Code that is never executed or used. This includes:
- Unreachable code after return statements
- Unused variables
- Unused functions
- Unused constants

Refactoring suggestion: Delete the dead code
"""

# Unused constant
THE_ANSWER_TO_EVERYTHING = 42


def format_currency(amount: float) -> str:
    """Unused function - never called"""
    return f"${amount:.2f}"


def active_function(value: int) -> int:
    if value < 0:
        return 0
        never_runs = value * -1  # Unreachable code after return
        print('This will never be printed', never_runs)

    temp = value * 2  # Unused variable

    return value + 1


def demo_dead_code() -> str:
    result = active_function(5)
    # format_currency is never called - dead code
    return format_currency(result)

File: java/src/main/java/com/refactoring/codesmells/dispensables/DeadCode.java
package com.refactoring.codesmells.dispensables;

public class DeadCode {

    // Constante no utilizada
    private static final int THE_ANSWER_TO_EVERYTHING = 42;

    public static void main(String[] args) {
        String result = demoDeadCode();
        System.out.println(result);
    }

    // Método no utilizado
    private static String formatCurrency(double amount) {
        return String.format("$%.2f", amount);
    }

    public static int activeFunction(int value) {
        if (value < 0) {
            return 0;
            // Código inalcanzable - nunca se ejecuta
            // int neverRuns = value * -1;
            // System.out.println("This will never be printed: " + neverRuns);
        }

        // Variable no utilizada
        int temp = value * 2;

        return value + 1;
    }

    public static String demoDeadCode() {
        int result = activeFunction(5);
        return formatCurrency(result);
    }
}
